src/CHIPBackend.cc

```diff
@@ -540,11 +542,20 @@ chipstar::Queue *chipstar::Device::getPerThreadDefaultQueue() {
   return getPerThreadDefaultQueueNoLock();
 }
 
+void chipstar::Device::QueueDeleter::operator()(
+    chipstar::Queue *q) const noexcept {
+  if (q) {
+    q->~Queue();
```

Comment:
What's going on here? Calling constructor directly and freeing the handle. Why not put the freeing to the Queue's destructor and delete it?